flang-rt/include/flang-rt/runtime/environment.h

```diff
@@ -42,6 +55,14 @@ struct ExecutionEnvironment {
       ExecutionEnvironment(){};
   void Configure(int argc, const char *argv[], const char *envp[],
       const EnvironmentDefaultList *envDefaults);
+
+  // Optional callback routines to be invoked pre and post
+  // execution environment setup.
+  void (*PreConfigureEnv)(int, const char *[], const char *[],
```

Comment:
Thank you, Dave!

It looks good to me, except maybe we should typedef the type of the callback functions (probably within `ExecutionEnvironment` struct), so that we can reduce text here and in the declaration of `RegisterConfigureEnv`.